api/src/main/java/io/opentelemetry/DefaultOpenTelemetry.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry;
+
+import static java.util.Objects.requireNonNull;
+
+import io.opentelemetry.baggage.BaggageManager;
+import io.opentelemetry.baggage.DefaultBaggageManager;
+import io.opentelemetry.baggage.spi.BaggageManagerFactory;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.propagation.DefaultContextPropagators;
+import io.opentelemetry.internal.Obfuscated;
+import io.opentelemetry.metrics.DefaultMeterProvider;
+import io.opentelemetry.metrics.Meter;
+import io.opentelemetry.metrics.MeterProvider;
+import io.opentelemetry.metrics.spi.MeterProviderFactory;
+import io.opentelemetry.trace.DefaultTracerProvider;
+import io.opentelemetry.trace.Tracer;
+import io.opentelemetry.trace.TracerProvider;
+import io.opentelemetry.trace.spi.TracerProviderFactory;
+import java.util.ServiceLoader;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * This class provides a static global accessor for telemetry objects {@link Tracer}, {@link Meter}
+ * and {@link BaggageManager}.
+ *
+ * <p>The telemetry objects are lazy-loaded singletons resolved via {@link ServiceLoader} mechanism.
+ *
+ * @see TracerProvider
+ * @see MeterProviderFactory
+ * @see BaggageManagerFactory
+ */
+@ThreadSafe
+final class DefaultOpenTelemetry implements OpenTelemetry {
+  private static final Object mutex = new Object();
+
+  static OpenTelemetry getGlobalOpenTelemetry() {
+    if (globalOpenTelemetry == null) {
+      synchronized (mutex) {
+        if (globalOpenTelemetry == null) {
+          globalOpenTelemetry = DefaultOpenTelemetry.builder().build();
+        }
+      }
+    }
+    return globalOpenTelemetry;
+  }
+
+  static void setGlobalOpenTelemetry(OpenTelemetry openTelemetry) {
+    globalOpenTelemetry = openTelemetry;
+  }
+
+  static Builder builder() {
+    return new Builder();
+  }
+
+  @Nullable private static volatile OpenTelemetry globalOpenTelemetry;
+
+  private final TracerProvider tracerProvider;
+  private final MeterProvider meterProvider;
+  private final BaggageManager baggageManager;
+
+  private final ContextPropagators propagators;
+
+  @Override
+  public TracerProvider getMyTracerProvider() {
```

Comment:
if we do merge this as-is, we need to get these renamed ASAP, for sure.

Reply:
can you stack a 2nd PR on top of this one with the renames, so we can move fast in getting it fixed?

Reply:
I'd set up another PR before merging this after it's green-lighted by @bogdandrutu